packages/webapp/src/components/Forms/Sale/index.jsx

```diff
@@ -1,85 +1,244 @@
-import Text from '../../Inputs/Text';
 import styles from './styles.module.scss';
-import { Control, Errors, Form } from 'react-redux-form';
-import DropDown from '../../Inputs/DropDown';
-import CropVarietySale from '../../Inputs/CropVarietySale';
-import React from 'react';
-import footerStyles from '../../LogFooter/styles.module.scss';
+import React, { useState, useEffect, useRef } from 'react';
 import { useTranslation } from 'react-i18next';
+import Form from '../../Form';
+import Button from '../../Form/Button';
+import Input, { getInputErrors } from '../../Form/Input';
+import Unit from '../../Form/Unit';
+import { useForm } from 'react-hook-form';
+import PageTitle from '../../PageTitle/v2';
+import FilterPillSelect from '../../Filter/FilterPillSelect';
+import { Error } from '../../Typography';
+import { harvestAmounts } from '../../../util/convert-units/unit';
+import { getDateInputFormat } from '../../../util/moment';
 
 const SaleForm = ({
-  currencySymbol,
   cropVarietyOptions,
-  chosenOptions,
-  handleChooseCrop,
-  model,
   onSubmit,
-  quantityUnit,
-  footerText,
-  footerOnClick,
+  onClickDelete,
+  title,
+  dateLabel,
+  customerLabel,
+  system,
+  currency,
+  sale,
 }) => {
+  // Reformat selector to match component format
+  // TODO: match component to selector format
```

Comment:
What does this mean?

Reply:
Similar to this:

Basically the selector returns an array of sales: 
`sales: {0: {sale 1, crop variety ID }, 1:{sale 2, crop variety ID}`

For css and hook form uniqueness reasons I wanted an array like:
`sales: {crop_variety_id: {sale 1, crop variety ID }, crop_variety_id: {sale 1, crop variety ID }}`

It was previously using something like:
`sales: {crop_variety_name: {sale 1, crop variety ID }, crop_variety_name: {sale 1, crop variety ID }}` 

I would like to work out a way to use the first syntax because then I can avoid extra code. But between the understand FilterPillSelect and other requirements I didn't get to it to meet this release deadline.

Reply:
Got it!